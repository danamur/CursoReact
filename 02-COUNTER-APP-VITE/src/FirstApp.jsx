export const FirstApp = () => {
  return (
    <>
        <h1>Daniel Namur</h1>
        <p>Desarrollador Full Stack</p>
    </>
  )
}