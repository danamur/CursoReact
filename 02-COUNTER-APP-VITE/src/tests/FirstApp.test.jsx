import { render } from '@testing-library/react'
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FirstApp />', () => {
    
    test('Debe de hacer match con el snapshot ', () => {
        
        const title = 'Bienvenido, Dwight';
        const { container } = render( <FirstApp title={ title }/>);

        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar el titulo en un h1', () => {
        
        const title = 'Bienvenido, Dwight';
        const { container, getByText } = render( <FirstApp title={ title }/>);

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);

    });

});