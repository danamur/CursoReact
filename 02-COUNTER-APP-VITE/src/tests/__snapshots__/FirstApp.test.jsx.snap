// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot  1`] = `
<div>
  <h1>
    Bienvenido, Dwight
  </h1>
  <p>
    123
  </p>
  <p>
    Daniel Namur
  </p>
</div>
`;
